### PR TITLE
Add apostrophe token filter page #7871

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
adding apostrophe token filter page

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7871

### Version
All

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _analyzers/token-filters/apostrophe-token-filter.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      "content": {
+        "type": "text",
+        "analyzer": "custom_analyzer"
+      }
+    }
+  }
+}
+```
+
+## Check generated tokens
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/apostrophe-token-filter.md:45]:
```suggestion
## Generated tokens
```

Comment (AntonEliatra) [reply _analyzers/token-filters/apostrophe-token-filter.md]: @kolchfa-aws Thank you for the feedback, thats now updated

File: _analyzers/token-filters/apostrophe.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+---
+
+# Apostrophe token filter
+
+The `apostrophe` token filter's primary function is to remove possessive apostrophes and anything following them. This can be very useful in analyzing text in languages which rely heavily on apostrophes, such as Turkish, where apostrophes serves to separate the root word from suffixes, including possessive suffixes, case markers, and other grammatical endings.
+
+
+## Example
+
+Following example can be used to create new index `custom_text_index` with custom analyzer configured in `settings` and used in `mappings`:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/apostrophe.md:15]:
```suggestion
The following example creates a new index named `custom_text_index` with a custom analyzer configured in `settings` and used in `mappings`:
```

File: _analyzers/token-filters/apostrophe.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      "start_offset": 34,
+      "end_offset": 38,
+      "type": "<ALPHANUM>",
+      "position": 6
+    }
+  ]
+}
+```
+
+The built in `apostrophe` token filter is not suitable for languages such as French, as the apostrophes are used at the beginning of the words, for example `"C'est l'amour de l'école"` will result in four tokens: "C", "l", "de", "l".
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/apostrophe.md:117]:
```suggestion
The built-in `apostrophe` token filter is not suitable for languages such as French, in which apostrophes are used at the beginning of the words. For example, `"C'est l'amour de l'école"` will result in four tokens: "C", "l", "de", "l".
```

**Review by kolchfa-aws — APPROVED** [review]
Thanks, @AntonEliatra! A couple of comments. Once addressed, I'll move this to editorial review.

**Pull request merged**